core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,123 @@
+    ]:
+        """Returns the Result object based on the validity of tags
+        in the exploration model.
+
+        Args:
+            exp: exp_models.ExplorationModel. The model of exploration
+                from which tags are fetched.
+
+        Returns:
+            Result[exp_models.ExplorationModel, str]. Result object that
+            contains ExplorationModel when the operation is successful
+            and str (containing error message) when invalid tags occurs.
+        """
+        tags = exp.tags
+        tags_are_valid = True
+        output_string = 'The exp of id %s contains' % exp.id
+        visited = set()
+        dup: List[str] = []
+        empty_tag = max_length_exceed_tag = 0
+
+        if len(tags) > 10:
+            tags_are_valid = False
+            output_string += ' tags length more than 10,'
+        if len(set(tags)) < len(tags):
+            tags_are_valid = False
+            for ele in tags:
+                if ele.strip() != '':
+                    if ele in visited or (visited.add(ele)):
+                        dup.append(ele)
+            if len(dup) != 0:
+                output_string += f' {len(dup)} duplicate values {dup},'
+
+        for tag in tags:
+            if tag.strip() == '':
+                tags_are_valid = False
+                empty_tag += 1
+            elif len(tag) > 30:
+                tags_are_valid = False
+                max_length_exceed_tag += 1
+
+        if (empty_tag != 0) or (max_length_exceed_tag != 0):
+            output_string += (
+                f' {empty_tag} empty tag and'
+                + f' {max_length_exceed_tag} tag having length more than 30, ')
+
+        last_comma_index = output_string.rfind(',')
+        output_string = (
+            output_string[:last_comma_index] + '.'
+            + output_string[last_comma_index + 1:])
+
+        if tags_are_valid:
+            return result.Ok(exp)
+        else:
+            return result.Err(output_string)
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns PCollection of explorations having invalid tags list
+        with the respective exploration id.
+
+        Returns:
+            PCollection. Returns PCollection of explorations having invalid
+            tags list with the respective exploration id.
+        """
+        total_explorations = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(include_deleted=False))
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_from_model)
+        )
+
+        exp_ids_with_invalid_tags = (
+            total_explorations
+            | beam.Map(self._check_tags_are_valid)
+        )
+
+        return (
+            exp_ids_with_invalid_tags
+            | 'Transform Results to JobRunResults' >> (
+                job_result_transforms.ResultsToJobRunResults())
+        )
```

Comment:
Should we print how many models don't have errors and how many have and then print the errors separately?

Reply:
Can you please provide some more info, like how to separate this Result's Pcollection. I tried and found https://github.com/oppia/oppia/blob/a8de2f7933f0f422e4edc3497761c389f093894f/core/jobs/transforms/job_result_transforms.py#L28
this class but I'm not sure how to use this in my case. It would be very helpful if you suggest something!!
Thanks!
@vojtechjelinek

Reply:
Hey, you can use this library https://github.com/rustedpy/result for generating either `Ok` or `Err` and then `ResultsToJobRunResults` for transforming it into outputable results. You can look into the other usages of `ResultsToJobRunResults` in our codebase to see how it can be used.